TrainingLoop/Callbacks/CSVLogger.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import ModelSupport
+
+/// A callback-based handler for logging the statistics to CSV file.
+public class CSVLogger {
+  public var path: String
+
+  let foundationFS: FoundationFileSystem
+  let foundationFile: FoundationFile
+
+  /// Create an instance that log statistics during the training loop.
+  public init(withPath path: String = "run/log.csv") {
```

Comment:
Almost never use "with" to introduce parameters.  It adds no information for the reader beyond what you'd get from `path`.  Of course you're calling this intializer “with” that path!

```suggestion
  public init(path: String = "run/log.csv") {
```

Reply:
Done.